tests/core/test_optimizations.py

```
@@ -0,0 +1,153 @@
+    """Fixture providing a clean CacheManager instance."""
+    return CacheManager(max_size=3)
+
+@pytest.fixture
+def performance_tracker():
+    """Fixture providing a clean PerformanceTracker instance."""
+    return PerformanceTracker()
+
+class TestCacheManager:
+    """Test suite for CacheManager class."""
+
+    def test_cache_initialization(self, cache_manager):
```

Ruff (0.8.2)

Missing return type annotation for public function `test_cache_initialization`

Add return type annotation: `None`

(ANN201)